### PR TITLE
optimisticLock

### PR DESCRIPTION
**Pull request by le-doude**
optimistic lock implementation

### CODE REVIEW

File: src/main/scala/io/enjapan/ops/TypedOperations.scala [hunk 1]
```diff
@@ -22,53 +28,71 @@ trait TypedOperations[M[_]] extends BucketOperations {
...
+                                            (fallback: T => T, retries: Int)
+                                            (implicit reads: Reads[T], writes: Writes[T], ec: ExecutionContext): M[Unit] = {
+
+    this.writeCas(t.id, t, cas)(fallback = fallback, retries)(reads, writes, ec)
+  }
+
+}
+
+object TypedOperations{
+  type CAS = Long
```
Comment (shawila) [inline src/main/scala/io/enjapan/ops/TypedOperations.scala]: Why not just put Long? Instead of CAS then CAS = Long? I R confuse.

Comment (le-doude) [inline src/main/scala/io/enjapan/ops/TypedOperations.scala]: this way the type is self explanatory

**Comment by shawila** [comment]
LGTM!
[LGTM](http://www.lgtm.in/i/cAHVU45Es)

[:+1:](http://www.lgtm.in/u/cAHVU45Es) [:-1:](http://www.lgtm.in/r/cAHVU45Es)

**Comment by pcboy** [comment]
LGTM

**Pull request merged**